packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -545,15 +555,23 @@ class FlutterPlatform extends PlatformPlugin {
       await Future.any<void>(<Future<void>>[
         testDevice.finished,
         () async {
-          final Uri? processVmServiceUri = await testDevice.vmServiceUri;
-          if (processVmServiceUri != null) {
-            globals.printTrace('test $ourTestCount: VM Service uri is available at $processVmServiceUri');
-          } else {
-            globals.printTrace('test $ourTestCount: VM Service uri is not available');
-          }
-          watcher?.handleStartedDevice(processVmServiceUri);
+          // These must be await-ed together, and if the first errors the second will never complete
```

Comment:
could you explain why `if the first errors the second will never complete`? setting `eagerError: true` implies this behavior, but it's not clear why this could happen

Reply:
Well, in real life, the most likely reason that the `remoteChannelCompleter.future` would error is because the app failed to start (for whatever reason). If the app fails to start, we're never gonna get a VM service URI, which is what `testDevice.vmServiceUri` is waiting for.

Reply:
at a high level, the trickiness here is because `testDevice.vmServiceUri` and `testDevice.start()` have dependencies between each other before they will both complete successfully, so if we fail someplace earlier we want to make sure we don't deadlock but handle it the error immediately (in particular, that the error bubble up from the right zone so the test runner knows there was a failure and it ends the test).